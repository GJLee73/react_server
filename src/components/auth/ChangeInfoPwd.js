import React, {Component} from 'react';
class ChangeInfoPwd extends Component {
    constructor(props) {
        super(props);
        this.state = { password:'',passwordCheck:'', pwdVerifyPhrase:'비밀번호는 8자이상 최소 한개 이상의 특수문자가 있어야합니다.',pwdCheckPhrase:'' }
        this.handleSend=this.handleSend.bind(this)
        this.handleChange=this.handleChange.bind(this)
        this.checkPwd=this.checkPwd.bind(this)
        this.verify=this.verify.bind(this)
    }
    checkPwd(){
		if(this.state.password !==this.state.passwordCheck || !this.state.password || !this.state.passwordCheck)
		{
			return this.setState({pwdCheckPhrase:'비밀번호가 다릅니다.'})
		}
		this.setState({pwdCheckPhrase:'비밀번호가 같습니다.',pwdCheck:true});
    }
    verify(e){
        var stringRegx = /[!@#$%^&*]/gi; 
        if(e.target.value.length>2){
            if(!stringRegx.test(e.target.value)){
                return this.setState({pwdVerifyPhrase:'최소 한개 이상의 특수문자가 포함되어야 합니다.(!@#$%^&*)'})
            }else if(e.target.value.length<8){
                return this.setState({pwdVerifyPhrase:'비밀번호가 너무 짧습니다.'})
            }
            this.setState({pwdVerifyPhrase:'안전한 비밀번호입니다.',pwdVerified:true})
        } 
    }
    handleChange(e){
        var pwd ={}
        pwd[e.target.name]=e.target.value

        this.setState(pwd)
    }
    handleSend(){
        var msg ={}
        msg.password=this.state.password
        this.props.submit(msg)
    }
    render() { 
        return (  <div className='card container'>
             <div className="input-field col s12">
        <label>Password </label>
        <i>{this.state.pwdVerifyPhrase}</i>
        <input
        name="password"
        type="password"
        className="validate"
        onChange={this.handleChange}
        value={this.state.password}
        onKeyUp={this.verify}/>
        </div>

        <div className="input-field col s12" >
            <label>Passsword check</label>
            <i>{this.state.pwdCheckPhrase}</i>
            <input
            name="passwordCheck"
            type="password"
            className="validate"
            onChange={this.handleChange}
            value={this.state.passwordCheck}
            onKeyUp={this.checkPwd}/>
           </div>
           <button onClick={this.handleSend}>완료</button>
        </div>  )
    }
}
 
export default ChangeInfoPwd;