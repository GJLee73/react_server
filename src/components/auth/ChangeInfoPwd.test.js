import React from 'react';
import {shallow} from 'enzyme'
import ChangeInfoPwd from './ChangeInfoPwd'

describe('ChangeInfoPwd test',()=>{
    var handleChange = jest.spyOn(ChangeInfoPwd.prototype,'handleChange')
    var handleSend = jest.spyOn(ChangeInfoPwd.prototype,'handleSend')
    var verify = jest.spyOn(ChangeInfoPwd.prototype,'verify')
    var checkPwd = jest.spyOn(ChangeInfoPwd.prototype,'checkPwd')
    var submit = jest.fn()
    var wrapper = shallow(<ChangeInfoPwd submit={submit}/>)

    var successEvent ={target:{value:'54375437!',name:'password'}}
    var failEvent ={target:{value:'1',name:'password'}}
    var failEvent1 ={target:{value:'123',name:'password'}}
    var failEvent2 ={target:{value:'5437!',name:'password'}}

    var failEvent_2nd ={target:{value:'a',name:'passwordCheck'}}
    var successEvent_2nd ={target:{value:'54375437!',name:'passwordCheck'}}

    var inputs= wrapper.find('input')
   
 
    describe('function test',()=>{
       it('handleChange should work',()=>{

            inputs.at(0).simulate('change',successEvent)
            inputs.at(1).simulate('change',successEvent_2nd)
      
            expect(handleChange.mock.calls.length).toBe(2)
            expect(wrapper.state()).toEqual({"password": "54375437!", "passwordCheck": "54375437!", "pwdCheckPhrase": "", "pwdVerifyPhrase": "비밀번호는 8자이상 최소 한개 이상의 특수문자가 있어야합니다."})
       })
       it('handleSend should work',()=>{
        wrapper.find('button').simulate('click')
        expect(handleSend.mock.calls.length).toBe(1)
        expect(submit.mock.calls.length).toBe(1)
       })
       it('verify test',async()=>{
        await inputs.at(0).simulate('change',failEvent)
        await inputs.at(0).simulate('keyup',failEvent)
        await expect(wrapper.state().pwdVerifyPhrase).toBe("비밀번호는 8자이상 최소 한개 이상의 특수문자가 있어야합니다.")
      
        await inputs.at(0).simulate('change',failEvent1)
        await inputs.at(0).simulate('keyup',failEvent1)
        expect(wrapper.state().pwdVerifyPhrase).toBe('최소 한개 이상의 특수문자가 포함되어야 합니다.(!@#$%^&*)')
   
        await inputs.at(0).simulate('change',failEvent2)
        await inputs.at(0).simulate('keyup',failEvent2)
        expect(wrapper.state().pwdVerifyPhrase).toBe('비밀번호가 너무 짧습니다.')

        await inputs.at(0).simulate('change',successEvent)
        await inputs.at(0).simulate('keyup',successEvent)
        expect(verify.mock.calls.length).toBe(4)


       })
        it('checkPwd test',()=>{
            inputs.at(1).simulate('change',failEvent_2nd)
            inputs.at(1).simulate('keyup',failEvent_2nd)
            expect(checkPwd.mock.calls.length).toBe(1)
            console.log(wrapper.state())
            expect(wrapper.state().pwdCheckPhrase).toBe('비밀번호가 다릅니다.')
            inputs.at(1).simulate('change',successEvent_2nd)
            inputs.at(1).simulate('keyup',successEvent_2nd)
            expect(wrapper.state().pwdCheckPhrase).toBe('비밀번호가 같습니다.')
           
        })

        

        

    })
    it('snapshot',()=>{
        expect(wrapper).toMatchSnapshot()
    })
})