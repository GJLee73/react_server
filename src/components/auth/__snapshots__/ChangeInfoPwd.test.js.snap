// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeInfoPwd test snapshot 1`] = `
<div
  className="card container"
>
  <div
    className="input-field col s12"
  >
    <label>
      Password 
    </label>
    <i>
      안전한 비밀번호입니다.
    </i>
    <input
      className="validate"
      name="password"
      onChange={[Function]}
      onKeyUp={[Function]}
      type="password"
      value="54375437!"
    />
  </div>
  <div
    className="input-field col s12"
  >
    <label>
      Passsword check
    </label>
    <i>
      비밀번호가 같습니다.
    </i>
    <input
      className="validate"
      name="passwordCheck"
      onChange={[Function]}
      onKeyUp={[Function]}
      type="password"
      value="54375437!"
    />
  </div>
  <button
    onClick={[Function]}
  >
    완료
  </button>
</div>
`;
